ideascube/mediacenter/templates/mediacenter/document_detail.html

```diff
@@ -18,7 +18,7 @@
     {% elif document.preview %}
         <a href="{% media document 'original' %}"><img src="{% media document 'preview' %}" /></a>
     {% endif %}
-    <div class="text">{{ document.summary }}</div>
+    <div class="text">{{ document.summary|safe }}</div>
```

Comment:
> But tinyMCE will send us valid html we don't want to escape. At first glance, I don't know how to make the distinction between valid html coming from tinyMCE and malicious one coming from evil user.

Thinking about it some more, what we want is not to escape the HTML: we want to limit the input to a set of known-safe HTML tags. `<p>`, `<a>`, `<h1>`, ... That sort of things. No `<script>`, basically.

This definitely looks like something we can do easily with BeautifulSoup, something like this:

```python
# ideascube/utils.py
from bs4 import (
    BeautifulSoup, CData, Comment, Declaration, Doctype, ProcessingInstruction,
)


TAG_WHITELIST = [
    'div', 'p', ...
]
SPECIAL_STRINGS = (CData, Comment, Declaration, Doctype, ProcessingInstruction)


def sanitize_html(html):
    soup = BeautifulSoup(html, 'html.parser')

    for tag in soup.find_all():
        if tag.name.lower() not in TAG_WHITELIST:
            tag.extract()
            continue

        # TODO: Maybe also sanitize the tag attributes?

    for string in soup.find_all(text=lambda t: isinstance(t, SPECIAL_STRINGS)):
        string.extract()

    return str(soup)
```

```python
# ideascube/widgets.py
class RichTextEditor(widgets.Widget):
    ...

    def value_from_datadict(self, data, files, name):
        return sanitize_html(data.get(name, ''))
```

This would avoid the security issue of someone sending malicious HTML directly as POST.

It does mean that we need to maintain the whitelist in accordance with whatever TinyMCE plugin we enable, and the risk is high that we one day enable a new plugin, and wonder for hours why it's not working, until we remember about the whitelist. But then that's probably better than a glaring security hole?